contrib/platform/src/com/sun/jna/platform/win32/User32.java

```diff
@@ -1950,4 +1956,192 @@ HDEVNOTIFY RegisterDeviceNotification(HANDLE hRecipient,
+	 *         <p/>
+	 *         If the function fails or times out, the return value is 0. To get
+	 *         extended error information, call GetLastError. If GetLastError
+	 *         returns ERROR_TIMEOUT, then the function timed out.
+	 *         <p/>
+	 *         Windows 2000: If {@link Kernel32#GetLastError()} returns 0, then
+	 *         the function timed out.
+	 */
+	long SendMessageTimeoutW(HWND hWnd, int msg, long wParam, long lParam,
+			int fuFlags, int uTimeout, DWORDByReference lpdwResult);
+	
+	/**
+	 * Retrieves the specified value from the WNDCLASSEX structure associated
+	 * with the specified window.
+	 * 
+	 * @param hWnd
+	 *            A handle to the window and, indirectly, the class to which the
+	 *            window belongs.
+	 * @param nIndex
+	 *            The value to be retrieved. To retrieve a value from the extra
+	 *            class memory, specify the positive, zero-based byte offset of
+	 *            the value to be retrieved. Valid values are in the range zero
+	 *            through the number of bytes of extra class memory, minus
+	 *            eight; for example, if you specified 24 or more bytes of extra
+	 *            class memory, a value of 16 would be an index to the third
+	 *            integer.To retrieve any other value from the WNDCLASSEX
+	 *            structure, specify one of the following values:
+	 *            {@link WinUser#GCW_ATOM} etc.
+	 * @return If the function succeeds, the return value is the requested
+	 *         value.
+	 *         <p/>
+	 *         If the function fails, the return value is zero. To get extended
+	 *         error information, call {@link Kernel32#GetLastError()}.
+	 */
+	long GetClassLongPtrA(HWND hWnd, int nIndex);
+	
+	/**
+	 * Retrieves the specified value from the WNDCLASSEX structure associated
+	 * with the specified window.
+	 * 
+	 * @param hWnd
+	 *            A handle to the window and, indirectly, the class to which the
+	 *            window belongs.
+	 * @param nIndex
+	 *            The value to be retrieved. To retrieve a value from the extra
+	 *            class memory, specify the positive, zero-based byte offset of
+	 *            the value to be retrieved. Valid values are in the range zero
+	 *            through the number of bytes of extra class memory, minus
+	 *            eight; for example, if you specified 24 or more bytes of extra
+	 *            class memory, a value of 16 would be an index to the third
+	 *            integer.To retrieve any other value from the WNDCLASSEX
+	 *            structure, specify one of the following values:
+	 *            {@link WinUser#GCW_ATOM} etc.
+	 * @return If the function succeeds, the return value is the requested
+	 *         value.
+	 *         <p/>
+	 *         If the function fails, the return value is zero. To get extended
+	 *         error information, call {@link Kernel32#GetLastError()}.
+	 */
+	long GetClassLongPtrW(HWND hWnd, int nIndex);
```

Comment:
You shouldn't need to create two different A and W versions of anything, without the extension it will default to the W version.